source/setup/components/SaveCredentialsPage.js

```diff
@@ -0,0 +1,304 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import { Async as Select } from "react-select";
+import FontAwesome from "react-fontawesome";
+import styled from "styled-components";
+import Spinner from "react-spinkit";
+import { Input as ButtercupInput, Button as ButtercupButton } from "@buttercup/ui";
+import { notifyError } from "../library/notify.js";
+import LayoutMain from "./LayoutMain.js";
+import { closeCurrentTab } from "../../shared/library/extension.js";
+import { FormButtonContainer, FormContainer, FormLegendItem, FormRow, FormInputItem } from "./forms.js";
+
+const FormInputItemColumnLayout = styled(FormInputItem)`
+    flex-direction: column;
+    justify-content: center;
+    align-items: flex-start;
+`;
+
+function flattenGroups(groups) {
+    const processed = [];
+    const nestGroup = (group, level = 0) => {
+        processed.push({
+            ...group,
+            level
+        });
+        group.groups.forEach(child => nestGroup(child, level + 1));
+    };
+    groups.forEach(group => nestGroup(group));
+    return processed;
+}
+
+const LoaderContainer = styled.div`
+    width: 100%;
+    height: 300px;
+    display: flex;
+    flex-direction: column;
+    justify-content: center;
+    align-items: center;
+`;
+const NoArchivesNotice = styled.span`
+    font-size: 13px;
+    color: rgb(242, 159, 4);
+`;
```

Comment:
Would be nice to have an empty line here

Reply:
I usually group constants by their type/context.. It should be the same in the other files. Simple values, styled components, required modules etc..

Reply:
ah it makes sense then

Reply:
☺️